deploy/terraform/network.tf

```diff
@@ -0,0 +1,303 @@
+  vpc_id = aws_vpc.nia_gp2gp_vpc.id
+  route {
+    cidr_block = "0.0.0.0/0"
+    gateway_id = aws_internet_gateway.nia_gp2gp_igw.id
+  }
+}
+
+resource "aws_route_table_association" "public" {
+  count          = var.subnet_count.public
+  route_table_id = aws_route_table.nia_gp2gp_public_rt.id
+  subnet_id      = aws_subnet.nia_gp2gp_public_subnet[count.index].id
+}
+
+resource "aws_route_table" "nia_gp2gp_private_rt" {
+  vpc_id = aws_vpc.nia_gp2gp_vpc.id
+}
+
+resource "aws_route_table_association" "private" {
+  count          = var.subnet_count.private
+  route_table_id = aws_route_table.nia_gp2gp_private_rt.id
+  subnet_id      = aws_subnet.nia_gp2gp_private_subnet[count.index].id
+}
+
+resource "aws_security_group" "nia_gp2gp_dmz" {
+  name        = "nia_gp2gp_api_sg"
+  description = "Public security group"
+  vpc_id      = aws_vpc.nia_gp2gp_vpc.id
+
+  ingress {
+    from_port   = var.http_server_port
+    protocol    = "tcp"
+    to_port     = var.http_server_port
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  //TODO: restrict access to match current restrictions
+
+  //ACCEPT TRAFFIC TO FACADE PORT 8081
+  ingress {
+    from_port = 8081
+    protocol  = "TCP"
+    to_port   = 8081
+    cidr_blocks = ["0.0.0.0/0"]
+    description = "Facade Traffic"
+  }
+
+  //ACCEPT TRAFFIC TO INBOUND PORT 443
+  ingress {
+    from_port = 443
+    protocol  = "TCP"
+    to_port   = 443
+    cidr_blocks = ["0.0.0.0/0"]
+    description = "Inbound Traffic"
+  }
+
+  //ACCEPT TRAFFIC TO MOCK SPINE PORT 8086
+  ingress {
+    from_port = 8086
+    protocol  = "TCP"
+    to_port   = 8086
+    cidr_blocks = ["0.0.0.0/0"]
+    description = "Mock Spine"
+  }
+
+  egress {
+    //allow all
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = {
+    Name = "nia_gp2gp_dmz_sg"
+  }
+}
+
+resource "aws_security_group" "nia_gp2gp_private" {
+  name        = "nia_gp2gp_private"
+  description = "Security group for private resources"
+  vpc_id = aws_vpc.nia_gp2gp_vpc.id
+
+  ingress {
+    description     = "Allow HTTPS traffic from only the pubic sg"
+    from_port       = "443"
+    protocol        = "tcp"
+    to_port         = "443"
+    security_groups = [aws_security_group.nia_gp2gp_dmz.id]
+  }
+
+  egress {
+    //allow all
+    from_port        = 0
+    to_port          = 0
+    protocol         = "-1"
+    cidr_blocks      = ["0.0.0.0/0"]
+    ipv6_cidr_blocks = ["::/0"]
+  }
+
+  tags = {
+    Name = "nia_gp2gp_db_sg"
+  }
+
+}
+
+resource "aws_security_group" "nia_gp2gp_mq" {
+  name        = "nia_gp2gp_mq"
+  description = "Security group for mq"
+  vpc_id = aws_vpc.nia_gp2gp_vpc.id
+
+  ingress {
+    description     = "Allow HTTP traffic from only the pubic sg"
+    from_port       = "5671"
+    protocol        = "tcp"
+    to_port         = "5671"
+    cidr_blocks      = ["0.0.0.0/0"]
+    ipv6_cidr_blocks = ["::/0"] //todo restrict traffic to mq properly
```

Comment:
Did you want to tackle this TODO?

Reply:
Sure, do we want to make a ticket for it?